src/sgx/faasm_sgx_native_symbols_wrapper.cpp

```diff
@@ -87,11 +121,92 @@ extern "C"
     {
         sgx_status_t sgxReturnValue;
         uint64_t returnValue;
-        if ((sgxReturnValue = ocall_faasm_read_state(
-               &returnValue, key, buffer_ptr, buffer_len)) != SGX_SUCCESS) {
+        uint32_t msg_payload_len, res_payload_len, encrypted_state_len;
```

Comment:
This file desperately needs some newlines and comments otherwise it's impossible to follow.